plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/Exception.hs

```diff
@@ -144,10 +146,18 @@ mapCauseInMachineException f = bimap (fmap (fmap f)) f
 
 -- | "Prismatically" throw an error and its (optional) cause.
 throwingWithCause
-    :: MonadError (ErrorWithCause e term) m
+    :: forall ex e t term m x
+    . (ex ~ ErrorWithCause e term, MonadError ex m)
```

Comment:
Why bind `ex` instead of just using `MonadError (ErrorWithCause e term) m`?

Reply:
So that it has a type parameter corresponding to `ex` which I can supply with `TypeApplications`, which seemed to be the easiest way to pin down the ambiguous bits of the types.

Reply:
Worth a comment then.